### DISCUSSION TITLE
how to set button disabled

### DISCUSSION THREAD

**Discussion by marsDes** [lit/lit#3734] [Q&A]
lit component
```ts
// t-button
@customElement('t-button')
...
return html`<button ?disabled=${this.disabled} @click=${console.log('button')}>
  <solt>
  </solt>
</button>`
...
```
use in html
```html
<t-button disabled onClick="console.log('t-button')">button</t-button>
```
it also log:'t-button',how can forbidden by add a `<slot></slot>`

**Comment by drk-mtr** [comment]
I'm not sure I understand the question. Maybe you could share a playground link with some comments in the code?

Do you mean [like this](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgc3RhdGV9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCAnLi94LWJ1dHRvbi5qcyc7XG5cbkBjdXN0b21FbGVtZW50KCdzaW1wbGUtZ3JlZXRpbmcnKVxuZXhwb3J0IGNsYXNzIFNpbXBsZUdyZWV0aW5nIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIEBzdGF0ZSgpXG4gIGlzQnV0dG9uRGlzYWJsZWQgPSBmYWxzZTtcbiAgXG4gIHRvZ2dsZURpc2FibGVkID0gKCkgPT4gdGhpcy5pc0J1dHRvbkRpc2FibGVkID0gIXRoaXMuaXNCdXR0b25EaXNhYmxlZDtcblxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgXG4gICAgICAgIDx4LWJ1dHRvbiA_ZGlzYWJsZWQ9XCIke3RoaXMuaXNCdXR0b25EaXNhYmxlZH1cIj48L3gtYnV0dG9uPlxuICAgICAgICA8YnV0dG9uIEBjbGljaz0ke3RoaXMudG9nZ2xlRGlzYWJsZWR9PlRvZ2dsZSBkaXNhYmxlZDwvYnV0dG9uPlxuICAgIGA7XG4gIH1cbn1cbiJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9zaW1wbGUtZ3JlZXRpbmcuanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxib2R5PlxuICA8c2ltcGxlLWdyZWV0aW5nIG5hbWU9XCJXb3JsZFwiPjwvc2ltcGxlLWdyZWV0aW5nPlxuPC9ib2R5PlxuIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9LHsibmFtZSI6IngtYnV0dG9uLnRzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgY3NzLCBMaXRFbGVtZW50fSBmcm9tICdsaXQnO1xuaW1wb3J0IHtjdXN0b21FbGVtZW50LCBwcm9wZXJ0eX0gZnJvbSAnbGl0L2RlY29yYXRvcnMuanMnO1xuXG5AY3VzdG9tRWxlbWVudCgneC1idXR0b24nKVxuZXhwb3J0IGNsYXNzIE15QnV0dG9uIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIEBwcm9wZXJ0eSh7IHR5cGU6IEJvb2xlYW4gfSlcbiAgZGlzYWJsZWQgPSBmYWxzZTtcbiAgXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgPGJ1dHRvbiA_ZGlzYWJsZWQ9JHt0aGlzLmRpc2FibGVkfT5NeSBidXR0b248L2J1dHRvbj5cbiAgICBgO1xuICB9XG59XG4ifV0)?

```ts
// simple-greeting.ts

import './x-button.js';

@customElement('simple-greeting')
export class SimpleGreeting extends LitElement {
  @state()
  isButtonDisabled = false;
  
  toggleDisabled = () => this.isButtonDisabled = !this.isButtonDisabled;

  render() {
    return html`
        <x-button ?disabled="${this.isButtonDisabled}"></x-button>
        <button @click=${this.toggleDisabled}>Toggle disabled</button>
    `;
  }
}

// x-button.ts

@customElement('x-button')
export class MyButton extends LitElement {
  @property({ type: Boolean })
  disabled = false;
  
  render() {
    return html`
        <button ?disabled=${this.disabled}>My button</button>
    `;
  }
}

**Comment by davidenke** [comment]
> ```ts
> // t-button
> @customElement('t-button')
> ...
> return html`<button ?disabled=${this.disabled} @click=${console.log('button')}>
>   <solt>
>   </solt>
> </button>`
> ...
> ```
>
> it also log:'t-button',how can forbidden by add a `<slot></slot>`

Looks like you have a typo at `<solt></solt>`. It should be `<slot></slot>`
